docs: added documentation about warnings (#5108)
fastify/fastify: docs/Reference/Warnings.md

<h1 align="center">Fastify</h1>

## Warnings

### Warnings In Fastify

> Warnings are enabled by default, and you can disable it by:

Reviewer: ```suggestion
Warnings are enabled by default, They can be disabled by using any of
the following methods:
```

- setting the `NODE_NO_WARNINGS` environment variable to `1`
- passing the `--no-warnings` flag to the node process

> - setting 'no-warnings' in the `NODE_OPTIONS` environment variable

Reviewer: Missing a link(s) to the core documentation about these options.

However, disabling warnings is not recommended as it may cause
potential problems when upgrading Fastify versions.
Only experienced users should consider disabling warnings.

### Fastify Warning Codes

| Code | Description | How to solve |
| ---- | ----------- | ------------ |
| **FSTWRN001** | The specified schema for a route is missing. This may indicate the schema is not well specified. | Check the schema for the route. |

> | **FSTDEP005** | You are accessing the deprecated `request.connection` property. | Use `request.socket`. |

Reviewer: I think warnings and deprecations should be listed in separate tables. Deprecations are slated to be removed in future versions. Warnings are permanent(ish).

It would also be good to link to the PR/Issue that introduced the warning/deprecation as the discussion can provide insight into the change and solution.

> | **FSTDEP006** | You are decorating Request/Reply with a reference type. This reference is shared amongst all requests. | Use `onRequest` hook. |

Reviewer 2: 'Use `onRequest` hook' is not the solution. It is about not using Arrays and Objects as values when decorating request and response.

> | **FSTDEP007** | You are trying to set a HEAD route using `exposeHeadRoute` route flag when a sibling route is already set. | Use `exposeHeadRoutes` from the server instance. |

Reviewer 2: > Use `exposeHeadRoutes` from the server instance.

Should be:
Remove `exposeHeadRoutes` or explicitly set `exposeHeadRoutes` to `false`

| **FSTDEP008** | You are using route constraints via the route `{version: "..."}` option.  |  Use `{constraints: {version: "..."}}` option.  |
| **FSTDEP009** | You are using a custom route versioning strategy via the server `{versioning: "..."}` option. |  Use `{constraints: {version: "..."}}` option.  |
| **FSTDEP010** | Modifying the `reply.sent` property is deprecated. | Use the `reply.hijack()` method. |

> | **FSTDEP011** | Variadic listen method is deprecated. | Use `.listen(optionsObject)`. | v4 |

Reviewer 2: Is this v4 extra?
```suggestion
| **FSTDEP011** | Variadic listen method is deprecated. | Use `.listen(optionsObject)`. |
```

Author: Yep, my bad. I'm pushing al the fixes

<h1 align="center">Fastify</h1>

## Warnings

### Warnings In Fastify
Warnings are enabled by default. They can be disabled by using any
of the following methods:

- setting the `NODE_NO_WARNINGS` environment variable to `1`
- passing the `--no-warnings` flag to the node process
- setting 'no-warnings' in the `NODE_OPTIONS` environment variable

For more information on how to disable warnings, see [node's documentation](https://nodejs.org/api/cli.html).

However, disabling warnings is not recommended as it may cause
potential problems when upgrading Fastify versions.
Only experienced users should consider disabling warnings.

### Fastify Warning Codes

| Code | Description | How to solve | Discussion |
| ---- | ----------- | ------------ | ---------- |
| **FSTWRN001** | The specified schema for a route is missing. This may indicate the schema is not well specified. | Check the schema for the route. | [#4647](https://github.com/fastify/fastify/pull/4647) |

> ### Fastify Deprecation Codes
> 
> | Code | Description | How to solve | Discussion |
> | ---- | ----------- | ------------ | ---------- |

Reviewer: ```suggestion
### Fastify Deprecation Codes

Deprecation codes are further supported by the Node.js CLI options:

- [--no-deprecation](https://nodejs.org/api/cli.html#--no-deprecation)
- [--throw-deprecation](https://nodejs.org/api/cli.html#--throw-deprecation)
- [--trace-deprecation](https://nodejs.org/api/cli.html#--trace-deprecation)

| Code | Description | How to solve | Discussion |
| ---- | ----------- | ------------ | ---------- |
```
